packages/toolpad-components/src/Text.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import {
+  Skeleton,
+  Typography as MuiTypography,
+  TypographyProps as MuiTypographyProps,
+  Link as MuiLink,
+  LinkProps as MuiLinkProps,
+  styled,
+} from '@mui/material';
+import { createComponent, TOOLPAD_COMPONENT_MODE_PROPERTY } from '@mui/toolpad-core';
+import ReactMarkdown from 'react-markdown';
+import remarkGfm from 'remark-gfm';
+
+type BaseProps = MuiLinkProps | MuiTypographyProps;
+interface TextProps extends Omit<BaseProps, 'children'> {
+  [TOOLPAD_COMPONENT_MODE_PROPERTY]: string;
+  value: string;
+  markdown: string;
+  href?: string;
+  loading?: boolean;
+}
+
+const MarkdownContainer = styled('div')({
+  display: 'block',
+  '&:empty::before': { content: '""', display: 'inline-block' },
+});
+
+function Text({
+  value,
+  markdown,
+  href,
+  loading,
+  [TOOLPAD_COMPONENT_MODE_PROPERTY]: mode,
+  sx,
+  ...rest
+}: TextProps) {
+  switch (mode) {
+    case 'markdown':
+      return loading ? (
```

Comment:
Will this result in layout shift? How dies the skeleton know which height to render? If we can't guarantee a loading state without layout shift, we shouldn't show a loading state imo

Reply:
For me, the utility of being able to show a loading state would trump the issue of it causing a layout shift. I would propose keeping the loading state, creating an issue for it and fixing it in an upcoming PR.